src/viam/sdk/rpc/dial.hpp

```diff
@@ -49,11 +56,15 @@ class DialOptions {
     const boost::optional<std::string>& entity() const;
     bool allows_insecure_downgrade() const;
     const std::chrono::duration<float>& timeout() const;
+    int initial_connection_attempts() const;
+    const std::chrono::duration<float>& initial_connection_attempt_timeout() const;
 
     void set_entity(boost::optional<std::string> entity);
     void set_credentials(boost::optional<Credentials> creds);
     void set_allow_insecure_downgrade(bool allow);
     void set_timeout(std::chrono::duration<float> timeout);
+    void set_initial_connection_attempts(int attempts);
```

Comment:
A small proposal: these and the other `set_x` methods should return `DialOptions&` so you can chain them:

```
DialOptions options{};
options.set_entity(...).set_timeout(...)....;
```